pkg/operator/controller/ingress/metrics.go

```diff
@@ -57,6 +57,17 @@ func SetIngressControllerConditionsMetric(ic *operatorv1.IngressController) {
 	}
 }
 
+// DeleteIngressControllerConditionsMetric deletes ingress_controller_conditions metrics which belong to the given ingresscontroller
+func DeleteIngressControllerConditionsMetric(ic *operatorv1.IngressController) {
+	for _, c := range ic.Status.Conditions {
+		if !reportedConditions.Has(c.Type) {
+			continue
+		}
+		deleted := ingressControllerConditions.DeleteLabelValues(ic.Name, string(c.Type))
+		log.V(4).Info("metrics deletion finished", "name_label", ic.Name, "condition_label", c.Type, "deleted", deleted)
```

Comment:
Might as well use phrasing and labels that are consistent with `SetIngressControllerConditionsMetric`:  
```suggestion
		log.V(4).Info("deleted metric for IngressController that is being deleted", "ingresscontroller", ic.Name, "condition_type", c.Type, "deleted", deleted)
```